starlette/staticfiles.py

```diff
@@ -226,8 +228,9 @@ def is_not_modified(
         try:
             if_none_match = request_headers["if-none-match"]
             etag = response_headers["etag"]
-            if if_none_match == etag:
-                return True
+            for match in ETAG_PATTERN.finditer(if_none_match):
+                if match.groups()[1] == etag:
+                    return True
```

Comment:
Looks good to me, though I prefer to avoid regexes where they aren't required, so we can possibly simplify this part.

```suggestion
            if etag in [tag.strip(r" W/") for tag in if_none_match.split(",")]:
                return True
```

Reply:
Good suggestion. This looks like more simplify. May I know why this string prefix with `r`. Is it to have some special meaning here?

Reply:
It's a typo, sorry :smile: